hadoop-common-project/hadoop-common/src/site/markdown/filesystem/fsdatainputstreambuilder.md

```diff
@@ -68,6 +68,11 @@ This is relevant with those stores which return version/etag information,
 including the S3A and ABFS connectors -they MAY use this to guarantee that
 the file they opened is exactly the one returned in the listing.
 
+There is no requirement for `status.getPath()` to the resolved path of
```

Comment:
Won't this cause uncertain failures?

Reply:
The reason for this is that viewfs and the hive fs wrappers have their own schemas/paths. If we insist on the paths matching, you woudn't be able to use the API through hive or mounted viewfs, which would be too brittle. Saying "it's just a hint" avoids this issue. Same reason I'm not worrying about type of the status.

s3a FS: use etag and version ID, if known
other status: use the length